# --- ROCK-COPYRIGHT-NOTE-BEGIN ---
# 
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# Please add additional copyright information _after_ the line containing
# the ROCK-COPYRIGHT-NOTE-END tag. Otherwise it might get removed by
# the ./scripts/Create-CopyPatch script. Do not edit this copyright text!
# 
# ROCK Linux: rock-src/package/base/ncurses/one-step-install.patch
# ROCK Linux is Copyright (C) 1998 - 2003 Clifford Wolf
# 
# This program is free software; you can redistribute it and/or modify
# it under the terms of the GNU General Public License as published by
# the Free Software Foundation; either version 2 of the License, or
# (at your option) any later version. A copy of the GNU General Public
# License can be found at Documentation/COPYING.
# 
# Many people helped and are helping developing ROCK Linux. Please
# have a look at http://www.rocklinux.org/ and the Documentation/TEAM
# file for details.
# 
# --- ROCK-COPYRIGHT-NOTE-END ---

--- ./mk-1st.awk.orig	2002-04-20 19:32:47.000000000 +0200
+++ ./mk-1st.awk	2003-11-10 09:04:39.651894512 +0100
@@ -52,8 +52,7 @@
 #
 function symlink(src,dst) {
 		if ( src != dst ) {
-			printf "rm -f %s; ", dst
-			printf "$(LN_S) %s %s; ", src, dst
+			printf "$(LN_S) -f %s %s; ", src, dst
 		}
 	}
 function rmlink(directory, dst) {
@@ -224,8 +223,7 @@
 					src_name = sprintf("../lib/%s", end_name);
 					dst_name = sprintf("$(DESTDIR)$(libdir)/%s", end_name);
 					printf "\t@echo installing %s as %s\n", src_name, dst_name
-					printf "\t-@rm -f %s\n", dst_name
-					printf "\t$(INSTALL_LIB) %s %s\n", src_name, dst_name
+					printf "\tinstall %s %s\n", src_name, dst_name
 
 					sharedlinks("$(DESTDIR)$(libdir)")
 
@@ -235,7 +233,7 @@
 				{
 					ovr_name = sprintf("libcurses%s", suffix)
 					printf "\t@echo linking %s to %s\n", end_name, ovr_name
-					printf "\tcd $(DESTDIR)$(libdir) && (rm -f %s; $(LN_S) %s %s; )\n", ovr_name, end_name, ovr_name
+					printf "\t$(LN_S) -f %s $(DESTDIR)$(libdir)/%s\n", end_name, ovr_name
 				}
 				if ( ldconfig != "" ) {
 					printf "\t- test -z \"$(DESTDIR)\" && %s\n", ldconfig
